sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_storage_utils.py

```diff
@@ -182,7 +183,12 @@ def get_artifact_path_from_storage_url(blob_url: str, container_name: dict) -> s
     return blob_url
 
 
-def get_ds_name_and_path_prefix(asset_uri: str) -> Tuple[str, str]:
-    ds_name = asset_uri.split("paths")[0].split("/")[-2]
-    path_prefix = asset_uri.split("paths")[1][1:]
+def get_ds_name_and_path_prefix(asset_uri: str, registry_name: str = None) -> Tuple[str, str]:
+    if registry_name: # for registry
+        split_paths = re.findall(STORAGE_URI_REGEX, asset_uri)
```

Comment:
This code puts a lot of faith that the structure of the asset URI is exactly matching the storage URI regex, how sure are we of this? 
Do we need to try catch with back up or at least try catch and throw a better error? Otherwise 
`path_prefix = split_paths[0][3]` will just say "index out of bounds" but we should throw error like "Asset ID is malformed"